src/advanced/JupyterNotebook/index.tsx

```diff
@@ -0,0 +1,327 @@
+   - write documentation 
+*/
+
+// cats an array of lines together
+const sourceLines = (lines: any[]) => {
+    return lines.reduce((result, value) => {
+        return result + value;
+    }, '');
+};
+
+export interface JupyterNotebookProps {
+    notebookData: string;
+}
+
+interface JupyterNotebookCellProps {
+    cell: any;
+    execution_count: string;
+    language: string;
+}
+
+interface JupyterNotebookOutputCellProps extends JupyterNotebookCellProps {
+    output: any;
+}
+
+interface JupyterNotebookCellRenderTemplateProps {
+    idColumnContent: string;
+    bodyColumnContent: any;
+    bodyColumnStyle: any;
+}
+
+const JupyterNotebookCellRenderTemplate: React.FC<JupyterNotebookCellRenderTemplateProps> = ({
+    idColumnContent,
+    bodyColumnContent,
+    bodyColumnStyle,
+}) => {
+    return (
+        <>
+            <Grid item xs={1}>
+                {idColumnContent}
+            </Grid>
+            <Grid item xs={11}>
+                <Paper variant={'outlined'} style={bodyColumnStyle}>
+                    {bodyColumnContent}
+                </Paper>
+            </Grid>
+        </>
+    );
+};
+
+// -- renders a cell of type 'markdown'
+const JupyterNotebookMarkdownCell: React.FC<JupyterNotebookCellProps> = ({ cell, execution_count, language }) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={''}
+            bodyColumnContent={<ReactMarkdown children={sourceLines(cell.source)} />}
+            bodyColumnStyle={{ padding: '0px 8px 0px 8px', margin: '0px 0px', border: '1px solid #AAA' }}
+        />
+    );
+};
+
+// -- renders a cell of type 'heading'
+const JupyterNotebookHeadingCell: React.FC<JupyterNotebookCellProps> = ({ cell, execution_count, language }) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={''}
+            bodyColumnContent={
+                <Container
+                    style={{
+                        marginBottom: '0px',
+                        boxShadow: 'none',
+                    }}
+                >
+                    <Heading variant="h1">{cell.source[0]}</Heading>
+                </Container>
+            }
+            bodyColumnStyle={{ padding: '0px 8px 0px 8px', margin: '0px 0px', border: '1px solid #AAA' }}
+        />
+    );
+};
+
+// -- renders a cell of type 'stream'
+const JupyterNotebookOutputStreamCell: React.FC<JupyterNotebookOutputCellProps> = ({
+    cell,
+    execution_count,
+    output,
+    language,
+}) => {
+    return (
+        <JupyterNotebookCellRenderTemplate
+            idColumnContent={`Out [${execution_count}]`}
+            bodyColumnContent={
+                <ReactMarkdown children={'**[' + output.name + ']**\n```\n ' + sourceLines(output.text) + '```'} />
+            }
+            bodyColumnStyle={{
+                padding: '0px 8px 0px 8px',
+                margin: '0px 0px',
+                backgroundColor: output.name === 'stderr' ? '#FDD' : '#DFD',
```

Comment:
Move Colors to const or perhaps add to northstar Color pallets

Reply:
sure